examples/lock-app/silabs/src/LockManager.cpp

```diff
@@ -475,18 +500,39 @@ bool LockManager::GetCredential(chip::EndpointId endpointId, uint16_t credential
     ChipLogProgress(Zcl, "Lock App: LockManager::GetCredential [credentialType=%u], credentialIndex=%d",
                     to_underlying(credentialType), credentialIndex);
 
-    const auto & credentialInStorage = mLockCredentials[to_underlying(credentialType)][credentialIndex];
+    uint16_t size = static_cast<uint16_t>(sizeof(LockCredentialInfo));
+
+    chip::StorageKeyName key = LockCredentialEndpoint(credentialIndex, credentialType, endpointId);
+
+    error = chip::Server::GetInstance().GetPersistentStorage().SyncGetKeyValue(key.KeyName(), &credentialInStorage, size);
+
+    // If no data is found at credential key
+    if (error == CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND)
+    {
+        // No credentials found
+        credential.status = DlCredentialStatus::kAvailable;
 
-    credential.status = credentialInStorage.status;
-    ChipLogDetail(Zcl, "CredentialStatus: %d, CredentialIndex: %d ", (int) credential.status, credentialIndex);
+        return true;
+    }
+    // Else if KVS read was successful
+    else if (error == CHIP_NO_ERROR)
+    {
+        credential.status = credentialInStorage.status;
+        ChipLogDetail(Zcl, "CredentialStatus: %d, CredentialIndex: %d ", (int) credential.status, credentialIndex);
+    }
+    else
+    {
+        ChipLogError(Zcl, "Error reading KVS key");
+        return false;
+    }
 
     if (DlCredentialStatus::kAvailable == credential.status)
     {
         ChipLogDetail(Zcl, "Found unoccupied credential ");
         return true;
     }
     credential.credentialType = credentialInStorage.credentialType;
-    credential.credentialData = credentialInStorage.credentialData;
+    credential.credentialData = chip::ByteSpan{ credentialInStorage.credentialData, credentialInStorage.credentialDataSize };
```

Comment:
And here again we commit to keep `uint8_t credentialData[kMaxCredentialSize];` Because `EmberAfPluginDoorLockCredentialInfo` is designed in a way that forces it.

I suggest we at least move this array to be static like mCredentials so we can only keep these two arrays to be static. Than being said, I am tempted to say that instead we should just refactor `EmberAfPluginDoorLockCredentialInfo` and `EmberAfPluginDoorLockUserInfo` so they they hold their own memories...

Otherwise we need to figure out a way to know when credential becomes out of scope.

Reply:
Any function callback that returns a span needs memory allocated somewhere. We could ask in slack if people are ok with refactoring EmberAfPluginDoorLockCredentialInfo and EmberAfPluginDoorLockUserInfo to hold their own storage